File: 27. Remove Element/27.RemoveElement.js
var removeElement = function (nums, val) {
  // Loop through the array using index i
  for (let i = 0; i < nums.length; i++) {
    // If the current element is equal to val, remove it
    if (nums[i] === val) {
      nums.splice(i, 1); // Remove the element at index i
      i--; // Decrement i to re-check the new element at index i
    }
  }
  return nums.length; // Return the new length of the array
};

File: 27. Remove Element/27.RemoveElement.java

class Solution {
    public int removeElement(int[] nums, int val) {
        int index = 0; // Keeps track of where to place the next valid element

        // Loop through the entire array
        for (int i = 0; i < nums.length; i++) {
            // If the current element is not equal to val, it is a valid element
            if (nums[i] != val) {
                nums[index] = nums[i]; // Place the valid element at index
                index++; // Move the index forward
            }
        }

        return index; // Return the new length of the array
    }
}